src/TwigComponent/src/ComponentPropertiesExtractor.php

```diff
@@ -0,0 +1,144 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\UX\TwigComponent;
+
+use Symfony\UX\TwigComponent\Attribute\ExposeInTemplate;
+use Symfony\UX\TwigComponent\Twig\PropsNode;
+use Twig\Environment;
+
+/**
+ * @author Jean-François Lépine <lepinejeanfrancois@gmail.com>
+ */
+final class ComponentPropertiesExtractor
+{
+    public function __construct(
+        private readonly Environment $twig,
+    ) {
+    }
+
+    /**
+     * @return array<array{display: string, name: string, type: string, default: mixed}>
```

Comment:
We can be a little bit more precise by setting `string` as the array key:
```suggestion
     * @return array<string, array{display: string, name: string, type: string, default: mixed}>
```